test/framework/src/main/java/org/elasticsearch/transport/AbstractSimpleTransportTestCase.java

```diff
@@ -158,7 +158,7 @@ public void setUp() throws Exception {
 
         final Settings connectionSettings = connectionSettingsBuilder.build();
 
-        serviceA = buildService("TS_A", version0, clusterSettingsA, connectionSettings); // this one supports dynamic tracer updates
+        serviceA = buildService("TS_A", CURRENT_VERSION, clusterSettingsA, connectionSettings); // this one supports dynamic tracer updates
```

Comment:
So, this change is a bit tricky. In these tests, the service was created with `CURRENT_VERSION.minimumCompatibleVersion()`. This was then called `minimumCompatibleVersion` *again* within `TransportHandshaker.sendHandshake`. This was causing the transport to initiate with the previous handshake version (6.8.0), with the older protocol.

In reality, it should be using CURRENT here, which is then turned into the `minimumCompatiblityVersion` *once*, resulting in version 7.17 (in this PR this is also replaced with the hardcoded `REQUEST_HANDSHAKE_VERSION`). This uses the new protocol, which has 4 extra bytes in the request & response.